umbral/curve.py

```diff
@@ -1,52 +1,69 @@
 from cryptography.hazmat.backends import default_backend
-from umbral import openssl
 
-_AVAIL_CURVES = {
-    'secp256r1': 415,
-    'secp256k1': 714,
-    'secp384r1': 715,
-}
+from umbral import openssl
 
 
 class Curve:
     """
     Acts as a container to store constant variables such as the OpenSSL
-    curve_nid, the EC_GROUP struct, and the order of the curve. This also acts
+    __curve_nid, the EC_GROUP struct, and the order of the curve. This also acts
     as a convenient whitelist to limit the curves used in pyUmbral.
     """
 
-    def __init__(self, curve_nid: int):
+    _supported_curves = {
+        'secp256r1': 415,
+        'secp256k1': 714,
+        'secp384r1': 715,
+    }
+
```

Comment:
Perhaps it's more correct wrt to how __init__ works to define the dict the other way around: NIDs as keys, names as values.

Reply:
:heavy_check_mark: